### DISCUSSION TITLE
What is the purpose of Tenant Identifier in tenants.json ?

### DISCUSSION THREAD

**Discussion by hung-doan** [OrchardCMS/OrchardCore#10161] [Q&A] [answered]
In my scenario, I backup a database from the remote server
- Then setup a fresh machine to start the development or debug on my local machine:
    -  Local File shell
    -  Multiple developers
    -  App_Data is excluded from the source version control. 
    - Single Tenant app
- Point the connectionString to the local database that I have restored from the backup file. 

However, I see that: 
- `If there is no tenants.json` in App_Data. It will ask us to go to the Setup screen. 
- Then I copied my tenants.json into the App_Data, all works. 

I also tried to change the `Identifier` in tenants.json. It stills works. 

```
{
  "Default": {
    "Identifier": ".......",
    "State": "Running"
  }
}
```

I wonder what is the purpose of `Identifier`, is that ok if I change it ?.

**Comment by jtkech** [comment]
@hung-doan 

Because of the following

https://github.com/OrchardCMS/OrchardCore/blob/8ff2453b72b3bcfc3e2c0181b220be5bc720fd96/src/OrchardCore/OrchardCore/Shell/ShellHost.cs#L219-L223

Notice that it is now named `VersionId`

**Reply by hung-doan** [reply]
cool, thanks you @jtkech. 
I dive a little more into the source code and see that these line of codes are executed via the setup process. 

That means the Identifier is just a flag that is used during the setup to avoid the race condition issue. 
The modification on this filed, logically doesn't cause any issues to our application ?

**Comment by jtkech** [comment, marked as answer]
Yes, used to keep in sync for a given OC instance the in memory id to the one in `tenants.json`. Note: If there are multiple OC running instances, these settings should be in a shared store e.g. the database, not in the local `tenants.json`.

Not only on setup, each time a given instance saves the settings, e.g. we can change a tenant prefix through the admin UI (with the tenants feature). No problem to change this Id if all the related OC instances are restarted.
